### PR TITLE
feat: configurable mailgun bulk email batch size #15725

### PR DESCRIPTION
**Pull request by jakub-k-slys**
Got some code for us? Awesome 🎊!

Please include a description of your change & check your PR against this list, thanks!

- [ ] There's a clear use-case for this code change, explained below
- [ ] Commit message has a short title & references relevant issues
- [ ] The build will pass (run `yarn test:all` and `yarn lint`)

We appreciate your contribution!

---

### <samp>🤖 Generated by Copilot at 7422ef5</samp>

This pull request adds a new configuration option for the Mailgun email provider that allows the user to set the maximum number of recipients per email batch. It also refactors the `MailgunEmailProvider` and `MailgunClient` classes to use this option and removes some unused code and tests.

### CODE REVIEW

**Comment by jakub-k-slys** [comment]
Hi @SimonBackx. I made changes according to your guidelines. Let me know your opinion. Thank you in advance!

File: ghost/mailgun-client/lib/MailgunClient.js [hunk 1]
```diff
@@ -305,4 +304,13 @@ module.exports = class MailgunClient {
...
         return !!instance;
     }
+
+    /**
+     * Returns configured batch size
+     *
+     * @returns {number}
+     */ 
+    getBatchSize() {
+        return this.#config.get('bulkEmail').get('batchSize');
```
[the remark this answers is not in the document]
Comment (SimonBackx) [reply ghost/mailgun-client/lib/MailgunClient.js:314]: As an alternative, 
```suggestion
        return this.#config.get('bulkEmail')?.batchSize ?? 1000;
```
can also work, and will probably require less changes in tests

**Comment by jakub-k-slys** [comment]
@SimonBackx thanks for your valuable feedback. I will fix those issues and get back to you.

**Comment by jakub-k-slys** [comment]
Hi @SimonBackx kindly ask for review :)

**Comment by jakub-k-slys** [comment]
Hi @SimonBackx, I fixed outstanding lint issue. Hope now is alright.

**Review by SimonBackx — APPROVED** [review]

**Comment by felipecruz91** [comment]
Hi, thanks for fixing this, @jakub-k-slys!

@jakub-k-slys @SimonBackx How can I set the batch size using environment variables? I'm using the Ghost Docker image.

**Comment by jakub-k-slys** [comment]
@felipecruz91 it is not set by the ENV variable, but by configuration file by adding:
...
"bulkEmail": {
  "batchSize": 666
}
...

**Pull request merged**